src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java

```diff
@@ -1,93 +1,82 @@
   private static final String EXPECTED_MEDREC_PATH = "src/test/resources/adsResources/recovered_medrec";
+  private static final String EXPECTED_SKY_PATH = "src/test/resources/adsResources/recovered_sky";
+  private static final By RECOVERABLE_SLOT_SELECTOR = By.cssSelector("[adonis-marker]");
+  private static final By PF_RECOVERED_ADS_SELECTOR = By.cssSelector("body>span");
 
   public AdsRecoveryObject(WebDriver driver, String page, Dimension resolution) {
     super(driver, page, resolution);
   }
 
-  public String getRecoveredAdUnitId(String adUnitId) {
-    String getIdScript = String.format("window._sp_.getElementId('%s')", adUnitId);
-    jsActions.waitForJavaScriptTruthy(String.format("window._sp_ && window._sp_.getElementId && %s", getIdScript));
-    return jsActions.execute(getIdScript).toString();
-  }
-
-  public void verifyPageFairRecoveryWithAdBlock() {
-    By spansBodyChildrenSelector = By.cssSelector("body>span");
-    Dimension topLeaderboardSize = new Dimension(728, 90);
-    Dimension medrecSize = new Dimension(300, 250);
+  public void verifyPageFairRecoveryWithAdBlock(int numberOfExpectedSlot) {
     String expectedRecoveredLB;
     String expectedRecoveredMR;
+    String expectedRecoveredSKY;
 
     try {
       expectedRecoveredLB = readFileToString(new File(EXPECTED_TOP_LEADERBOARD_PATH));
       expectedRecoveredMR = readFileToString(new File(EXPECTED_MEDREC_PATH));
+      expectedRecoveredSKY = readFileToString(new File(EXPECTED_SKY_PATH));
     } catch (IOException e) {
       PageObjectLogging.log("Can't open expected PageFair recovery file.", e, false);
       throw new WebDriverException("Can't open expected PageFair recovery file.");
     }
 
     // when PF recovered ad is on page, inserts span elements as a direct children of body
-    wait.forElementPresent(spansBodyChildrenSelector);
+    wait.forElementPresent(PF_RECOVERED_ADS_SELECTOR);
 
     // verify that adblock is turned on on that page
     verifyNoAdsOnPage();
 
-    String firstSpanClass = driver.findElement(spansBodyChildrenSelector).getAttribute("class");
-    List<WebElement> recoveredAds = driver
-        .findElements(By.cssSelector("body>span." + firstSpanClass))
-        .stream()
-        .filter(WebElement::isDisplayed)
-        .filter(e -> e.getCssValue("background").contains("data:image/jpeg"))
-        .collect(Collectors.toList());
+    List<WebElement> recoveredAds = getRecoveredAds(PF_RECOVERED_ADS_SELECTOR);
 
-    Assert.assertEquals(recoveredAds.size(), RECOVERABLE_ADS_COUNT);
+    Assert.assertEquals(recoveredAds.size(), numberOfExpectedSlot);
 
     for (WebElement ad : recoveredAds) {
       Dimension adSize = ad.getSize();
 
-      if (adSize.equals(topLeaderboardSize)) {
+      if (adSize.equals(TOP_LEADERBOARD_SIZE)) {
         Assertion.assertTrue(ad.getCssValue("background").contains(expectedRecoveredLB), "TOP_LEADERBOARD is not correctly recovered!");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "background" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)